### ISSUE TITLE
Got "Internal Server Error" when browsing https://localhost:4443/

### ISSUE THREAD

**Issue by FNakano** [AtomGraph/LinkedDataHub#134]
Hello,
    I'm installing LinkedDataHub.
    Followed the instructions in https://atomgraph.github.io/LinkedDataHub/linkeddatahub/docs/get-started/setup/. The `docker-compose up --build ` command exited with a permission denied error which was overcome by prepending `sudo`. Is it ok, or, is it expected?
    It looks that the webserver is up (Fuseki with Tomcat?). 
    I navigated to https://localhost:4443/ using the certificate generated by `./scripts/setup.sh .env ssl $owner_cert_pwd $secretary_cert_pwd 3650 ` and imported into Firefox as instructed in https://atomgraph.github.io/LinkedDataHub/linkeddatahub/docs/get-started/get-an-account/. Got the messages in the screen capture below:

![Captura de tela de 2022-10-02 17-04-57](https://user-images.githubusercontent.com/14809968/193473999-243cc7a3-96e6-4f25-87e9-f4ad8a532a3d.png)

Messages in the terminal running `sudo docker-compose up --build `:

<pre><font color="#A347BA">linkeddatahub_1     |</font> 02-Oct-2022 21:31:44.449 INFO [main] org.apache.catalina.startup.HostConfig.deployDescriptor Deployment of deployment descriptor [/usr/local/tomcat/conf/Catalina/localhost/ROOT.xml] has finished in [3,457] ms
<font color="#A347BA">linkeddatahub_1     |</font> 02-Oct-2022 21:31:44.451 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler [&quot;http-nio-7070&quot;]
<font color="#A347BA">linkeddatahub_1     |</font> 02-Oct-2022 21:31:44.455 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [3477] milliseconds
^[[A^[[A^[[A^[[B^[[B<font color="#12488B">nginx_1             |</font> 172.18.0.1 - - [02/Oct/2022:19:40:59 +0000] &quot;GET / HTTP/1.1&quot; 500 1012 &quot;https://atomgraph.github.io/&quot; &quot;Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0&quot;
<font color="#12488B">nginx_1             |</font> 172.18.0.1 - - [02/Oct/2022:19:40:59 +0000] &quot;GET /favicon.ico HTTP/1.1&quot; 404 774 &quot;https://localhost:4443/&quot; &quot;Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0&quot;
<font color="#12488B">nginx_1             |</font> 172.18.0.1 - - [02/Oct/2022:19:41:28 +0000] &quot;GET / HTTP/1.1&quot; 500 1012 &quot;-&quot; &quot;Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0&quot;

</pre>

How can I fix it?

**Comment by namedgraph** [comment]
Hi @FNakano. Looks like something's wrong with the config file. The `sudo` part is weird, but LinkedDataHub should be able to run under a non-root user.
Some steps we can try:
* Can you send me the full console log?
* Can you show me your `.env` file? Scrub the private info if any.
* Try `docker exec $container_id cat /var/linkeddatahub/datasets/system.trig` and paste what you get. Replace `$container_id` with LinkedDataHub's container ID first.

**Comment by FNakano** [comment]
Hi, @namedgraph . 

All the names and password were created to try out LinkedDataHub. If any name/password is expected match users in the computer LinkedDataHub is installed, there could cause an error.

Here is the `.env` complete listing:

```
COMPOSE_CONVERT_WINDOWS_PATHS=1
COMPOSE_PROJECT_NAME=linkeddatahub

PROTOCOL=https
HTTP_PORT=81
HTTPS_PORT=4443
HOST=localhost
ABS_PATH=/home/fabio

# owner's WebID certificate properties
OWNER_MBOX=fabio@localhost
OWNER_GIVEN_NAME=Fabio
OWNER_FAMILY_NAME=Doe
OWNER_ORG_UNIT=Unique
OWNER_ORGANIZATION=Me
OWNER_LOCALITY=SomeCity
OWNER_STATE_OR_PROVINCE=SomeState
OWNER_COUNTRY_NAME=SomeCountry

SECRETARY_CERT_PASSWORD=H&ureka5!

SECRETARY_CERT_PASSWORD=fabio

SECRETARY_CERT_PASSWORD=H&ureka5!

SECRETARY_CERT_PASSWORD=H&ureka5!
```

I don't remember to have written these four `SECRETARY_CERT_PASSWORD` lines. I actually inserted the values while generating ssl certificates. I believe these lines were inserted in `.env` in the build process.

I didn´t record that full console log. It is lost. I can generate another one. Just send me instructions.

Where can I get the $container_id ?

Thanks,
Fábio.

**Comment by namedgraph** [comment]
You can get container ID by looking at the output of `docker ps`.

I think I know what the problem is though -- you have configured the system base URI in the `.env` file as `https://localhost:4443/home/fabio`, and application base URIs are resolved against that.  But you're attempting to open `https://localhost:4443/` instead.

Try leaving the `ABS_PATH` as in the example, i.e. `ABS_PATH=/`.
Otherwise you would need to use `ABS_PATH=/home/fabio/` and change the [`system-varnish.trig` config file](https://github.com/AtomGraph/LinkedDataHub/blob/master/config/system-varnish.trig#L37) that contains the application definitions to contain `ldt:base <home/fabio/>` (for the `end-user` app) and `ldt:base <home/fabio/admin/>` (for the `admin` app).

**Comment by FNakano** [comment]
Hi, @namedgraph .
   It works!
   I followed your instructions about `.env` file: (leave the ABS_PATH as in the example).
   Then I did some silly thing: erase the local copy of LinkedDataHub and clone (the forked repo) again.
   Followed the setup instructions, replaced the certificate in the browser.
   Got the initial screen:
![Captura de tela de 2022-10-05 13-05-49](https://user-images.githubusercontent.com/14809968/194109731-4aa64843-f08e-4af4-8a71-ee3b6f7ae10b.png)
   Now I can go further.
Thank you!
